cloudwatch/cloudwatch.go

```diff
@@ -650,17 +650,17 @@ func (output *OutputPlugin) putLogEvents(stream *logStream) error {
 				logrus.Errorf("[cloudwatch %d] Encountered error %v; detailed information: %s\n", output.PluginInstanceID, awsErr, awsErr.Message())
 				if strings.Contains(awsErr.Message(), "group") {
 					if err := output.createLogGroup(&Event{group: stream.logGroupName}); err != nil {
-						logrus.Error(err)
+						logrus.Errorf("[cloudwatch %d] Encountered error %v\n", output.PluginInstanceID, err)
 						return err
 					}
 				} else if strings.Contains(awsErr.Message(), "stream") {
 					if _, err := output.createStream(&Event{group: stream.logGroupName, stream: stream.logStreamName}); err != nil {
-						logrus.Error(err)
+						logrus.Errorf("[cloudwatch %d] Encountered error %v\n", output.PluginInstanceID, err)
 						return err
```

Comment:
Keep the log messages as is- we always include the plugin ID in all log statements.

The problem before was that you added the plugin ID in the `fmt.Errorf` which creates an error. When that error is printed by the calling function the plugin ID would be printed twice since it would both be in the error message and in the log statement.